Use sphinxcontrib-spelling instead of aspell

tobsan opened the pull request:
There are some hickups when it comes to using several word list files,
but the maintainer of the plugin is open to patches!

Depends on [SWF-blueprint/#88](https://github.com/Pelagicore/software-factory-blueprint/pull/88)

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

Commit:
Use sphinxcontrib-spelling instead of aspell

There are some hickups when it comes to using several word list files,
but the maintainer of the plugin is open to patches!

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

chuanjin commented:
README also needs to be updated

demonicblue reviewed (approved):
Looks good. Are you going to update the readme in this PR?

tobsan commented:
I really should update the README, yes. Will do!

tobsan commented:
README updated!

demonicblue commented:
Add spellcheck to the dependency section of the README as well.

tobsan commented:
@demonicblue good call, added.

demonicblue commented:
It should also be added to the pip install command.
(Sorry for all the back-and-forth :sweat_smile:)

tobsan committed:
Use the multi-wordlist spelling option in sphinx

A recent change in sphinxcontrib-spelling allows for several word lists
with known spellings of words. We should use that instead of the CMake
hack.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>
